File: Problem005/java/Main.java
//nlog(logn)‚ÌƒI[ƒ_‚Å‰ğ‚¢‚½
//232792560
class Main{
	public static void main(String[] args){
		System.out.println(gcd(12, 10));
		int max = 20;
		long n = 1;
		for(int i=2;i<=max;i++){
			n = (n * i) / gcd(n, i);
		}
		System.out.println(n);
	}
	
	static long gcd(long a, long b){
		if(a *b == 0)return Math.max(a, b);
		if(a > b){
			return gcd(a%b, b);
		}else{
			return gcd(a, b%a);
		}
	}
}

File: Problem005/go/main.go
package main

import (
	"fmt"
)

func gcd(m, n int) int {
	if n == 0 {
		return m
	}
	return gcd(n, m%n)
}

func main() {
	sum := 1
	for i := 1; i < 21; i++ {
		sum = (sum * i) / gcd(sum, i)
	}
	fmt.Println(sum)
}
